src/docker/remote.rs

```diff
@@ -561,7 +564,7 @@ impl QualifiedToolchain {
             .file_name()
             .expect("should be able to get toolchain name")
             .to_utf8()?;
-        let toolchain_hash = path_hash(self.get_sysroot())?;
+        let toolchain_hash = path_hash(self.get_sysroot(), 5)?;
```

Comment:
could we make the path static variables?

Reply:
Sure. I was going to add a comment on the PR that I made the existing ones all 5, but for the mount identifier, I made it require 10 hexdigits. 

For the old hashes, since they either are for the toolchain path (very few actual paths, especially with rustup), a unique ID for the container (which is unique because of the system time, so a small hash is fine), the package path (which is also prefixed by the package name, so small number of possibilities in real cases), 16^5 or ~1million is plenty unique. 

But we run into a birthday problem pretty fast (we're using SHA1 so it's all trusted data) if we consider a user mounting many directories or a large number of users mounting a non-trivial number of directories each. If we have 16^5 options, asuming everything is perfectly distributed, that means we have a 50% chance of collision at ~1200 directories. If we up that to 10 hexdigits, then we have no issue.

We shouldn't modify the toolchain ID, since it would mean existing data volumes would no longer be used, and there's no reason to modify them. But for arbitrary volume mounts, it's probably a good idea to ensure there's no accidental collision.

Reply:
I've added the following global const variables, so it's clear why we're using each in a given scenario.

```rust
/// Short hash for identifiers with minimal risk of collision.
pub const PATH_HASH_SHORT: usize = 5;

/// Longer hash to minimize risk of random collisions
/// Collision chance is ~10^-6
pub const PATH_HASH_UNIQUE: usize = 10;
```